kitsune/src/http/pagination.rs

```diff
@@ -0,0 +1,71 @@
+use std::{convert::Infallible, fmt::Display};
+
+use axum::{
+    response::{IntoResponseParts, ResponseParts},
+    Json,
+};
+use http::HeaderValue;
+
+pub type PaginatedJsonResponse<T> = (Option<Link<Vec<(&'static str, String)>>>, Json<Vec<T>>);
+
+pub struct Link<T>(pub T);
+
+impl<T, K, V> IntoResponseParts for Link<T>
+where
+    T: IntoIterator<Item = (K, V)>,
+    K: Display,
+    V: Display,
+{
+    type Error = Infallible;
+
+    fn into_response_parts(self, mut res: ResponseParts) -> Result<ResponseParts, Self::Error> {
+        let value = self
+            .0
+            .into_iter()
+            .map(|(key, value)| format!("<{value}>; rel=\"{key}\""))
+            .collect::<Vec<String>>()
+            .join(", ");
+
+        // TODO: log an error when HeaderValue conversion fails
```

Comment:
Can't we change the associated error type to `InvalidHeaderValue`? Are there trait bounds that aren't satisfied?

Reply:
Oh, it wants an `IntoResponse` error. I think you can just use our error "super enum" for now. That should implement `From<InvalidHeaderValue>` and `IntoResponse`

Reply:
Oh yeah, that works, thanks! I just had to map the error to `http`'s `Error` first